File: Practico4/P4Ejercicio9/Java/vista/Vista.java
import javax.swing.*;
import javax.swing.table.DefaultTableModel;
import java.awt.*;

public class Vista extends JFrame {

    private JButton botonSeleccionarArchivo;
    private JButton botonLeerArchivo;
    private JTable tabla;
    private JLabel mensajeError;
    private DefaultTableModel tableModel;

    public Vista() {
        // Configuración de la ventana principal
        setTitle("Lector CSV");
        setSize(600, 400);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        // Configuración del layout principal
        JPanel panelCentral = new JPanel();
        panelCentral.setLayout(new BorderLayout());

        JPanel layoutBotones = new JPanel();
        layoutBotones.setLayout(new FlowLayout());

        // Botón de seleccionar archivo
        botonSeleccionarArchivo = new JButton("Seleccionar Archivo CSV");
        layoutBotones.add(botonSeleccionarArchivo);

        // Botón de leer archivo
        botonLeerArchivo = new JButton("Leer Archivo");
        layoutBotones.add(botonLeerArchivo);

        // Añadir botones al panel principal
        panelCentral.add(layoutBotones, BorderLayout.NORTH);

        // Configuración de la tabla
        tableModel = new DefaultTableModel();
        tabla = new JTable(tableModel);
        tableModel.setColumnIdentifiers(DataLabels.LABELS.toArray());  // Configurar encabezados de columna
        
        JScrollPane scrollPane = new JScrollPane(tabla);
        panelCentral.add(scrollPane, BorderLayout.CENTER);

        // Mensaje de error
        mensajeError = new JLabel("");
        mensajeError.setForeground(Color.RED);
        panelCentral.add(mensajeError, BorderLayout.SOUTH);

        // Añadir el panel al marco
        add(panelCentral);
    }

    public JButton getBotonSeleccionarArchivo() {
        return botonSeleccionarArchivo;
    }

    public JButton getBotonLeerArchivo() {
        return botonLeerArchivo;
    }

    public JTable getTabla() {
        return tabla;
    }

    public JLabel getMensajeError() {
        return mensajeError;
    }

    public DefaultTableModel getTableModel() {
        return tableModel;
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            Vista vista = new Vista();
            vista.setVisible(true);
        });
    }
}

File: P4Ejercicio9/Python/vista/Vista.py
from PyQt6.QtWidgets import QMainWindow, QWidget, QVBoxLayout, QPushButton, QTableWidget, QTableWidgetItem, QFileDialog, QLabel, QHBoxLayout

from modelo.DataLabels import DataLabels

class Vista(QMainWindow):
    def __init__(self):
        super().__init__()
        self.setWindowTitle("Lector CSV")
        self.setGeometry(100, 100, 600, 400)

        self.central_widget = QWidget(self)
        self.setCentralWidget(self.central_widget)

        self.layout_principal = QVBoxLayout()
        self.layout_botones = QHBoxLayout()

        self.boton_seleccionar_archivo = QPushButton("Seleccionar Archivo CSV")
        self.layout_botones.addWidget(self.boton_seleccionar_archivo)

        self.boton_leer_archivo = QPushButton("Leer Archivo")
        self.layout_botones.addWidget(self.boton_leer_archivo)

        self.tabla = QTableWidget()
        self.tabla.setColumnCount(len(DataLabels.LABELS))
        self.tabla.setHorizontalHeaderLabels(DataLabels.LABELS)

        self.mensaje_error = QLabel("")
        self.mensaje_error.setStyleSheet("color: red;")

        self.layout_principal.addLayout(self.layout_botones)
        self.layout_principal.addWidget(self.tabla)
        self.layout_principal.addWidget(self.mensaje_error)

        self.central_widget.setLayout(self.layout_principal)
